modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4777,6 +4777,83 @@ Point LineIterator::pos() const
 
 //! @endcond
 
+class CV_EXPORTS LineVirtualIterator
+{
+public:
+    /** @brief initializes the iterator
+
+    creates iterators for the line connecting pt1 and pt2
+    the line will be clipped on the image boundaries
+    the line is 8-connected or 4-connected
+    If leftToRight=true, then the iteration is always done
+    from the left-most point to the right most,
+    not to depend on the ordering of pt1 and pt2 parameters
+    */
+    LineVirtualIterator(const Rect& aRect, const Point& pt1, const Point& pt2,
+                        int connectivity = 8, bool forceLeftToRight = false ) {
+      currentPosOffset = Point(aRect.x, aRect.y);
+      init(aRect.size(), pt1, pt2, connectivity, forceLeftToRight);
+    }
+    LineVirtualIterator(const Size& aSize, const Point& pt1, const Point& pt2,
+                        int connectivity = 8, bool forceLeftToRight = false ) {
+      init(aSize, pt1, pt2, connectivity, forceLeftToRight);
+    }
+    LineVirtualIterator(const Point& pt1, const Point& pt2,
+                        int connectivity = 8, bool forceLeftToRight = false );
+    void init(const Size& size, const Point& pt1, const Point& pt2,
```

Comment:
What is the purpose of public `init` method?

All initialization stuff can be done by constructors chaining.
Your main constructor is `LineVirtualIterator(const Size&, const Point&, const Point&, int, bool)`
```c++
struct A {
    A(int arg1, int arg2, int arg3) {
        std::cout << "Initialized with " << arg1 << ' ' << arg2 << ' ' << arg3 << '\n';
    }

    A(int arg1, int arg2) : A(arg1, arg2, 123) {}

    explicit A(int arg1) : A(arg1, 4) {}
};
```

Also note, that sometimes it is reasonable to use a ["named" constructor idiom](https://isocpp.org/wiki/faq/ctors#named-ctor-idiom) to make user understanding clear, what object is going to be created. (RVO eliminates copy and move constructor calls)

Reply:
As an alternative you give better names to parameters instead of `aSize`/`aRect`